### PR TITLE
fix(c-api) `wasi_version_t` wasn't bound correctly to C

### PR DESCRIPTION
**Pull request by Hywan**
# Description

Because the enum variants weren't in uppercase _and_ because one variant (`InvalidVersion`) has a non-literal value, `cbindgen` was ignoring the variants. This patch fixes that. It's not a breaking changes since the WASI version constants weren't present in the `wasmer_wasm.h` file before.

Note: `u32::max_value()` is soft-deprecated, we should use `u32::MAX` instead. Notice that none of them works with `cbindgen`, so I've hardcoded the value of `u32::MAX` here.

# Review

- [x] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
 - [#2056](https://github.com/wasmerio/wasmer/pull/2056) Change back to depend on the `enumset` crate instead of `wasmer_enumset`
 
 ### Fixed
+- [#2058](https://github.com/wasmerio/wasmer/pull/2058) `wasm_version_t` wasn't bound correctly to C.
```
Comment (MarkMcCaskey) [inline CHANGELOG.md:19]: This message in the changelog isn't meaningful without context, perhaps change it to something that users can understand as it affects them

Comment (MarkMcCaskey) [inline CHANGELOG.md:19]: This message in the changelog isn't meaningful without context, perhaps change it to something that users can understand as it affects them

**Review by MarkMcCaskey — APPROVED** [review]

**Review by jubianchi — APPROVED** [review]

**Review by syrusakbary — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Hywan) [reply CHANGELOG.md:19]:
```suggestion
- [#2058](https://github.com/wasmerio/wasmer/pull/2058) Expose WASI versions to C correctly.
```

**Comment by Hywan** [comment]
Error on Windows (for the record):

```
D:\a\wasmer\wasmer\tests\integration\cli/../../../lib/c-api\wasmer_wasm.h:105:18: error: typedef redefinition with different types ('uint32_t' (aka 'unsigned int') vs 'wasi_version_t')
    typedef uint32_t wasi_version_t;
                     ^
    D:\a\wasmer\wasmer\tests\integration\cli/../../../lib/c-api\wasmer_wasm.h:70:6: note: previous definition is here
    enum wasi_version_t {
```

**Comment by Hywan** [comment]
https://github.com/wasmerio/wasmer/pull/2058/commits/7a435ef740b4c31e8662261b1ad67e43d2f8d0ad

> On Windows, using a `u32` representation for `wasi_version_t` fails if
a C++ compiler is used to treat a C program. So we change our strategy
here. We use a C representation to be FFI-safe, and the
`INVALID_VERSION` variant is now set to -1 instead of `u32::MAX`.
> 
> This patch also adds unit tests for `wasi_get_wasi_version` so that we
are sure of the behavior of this `type` on all platforms.

**Pull request merged**